vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1713,7 +1720,21 @@ def execute_model(
         max_gen_len = sampled_token_ids.shape[-1]
         if max_gen_len == 1:
             # No spec decode tokens.
-            valid_sampled_token_ids = sampled_token_ids.tolist()
+
+            # This is a short term mitigation for issue mentioned in
+            # https://github.com/vllm-project/vllm/issues/22754.
+            # `tolist` would trigger a cuda wise stream sync, which
+            # would block other copy ops from other cuda streams.
+            # A cuda event sync would avoid such a situation. Since
+            # this is in the critical path of every single model
+            # forward loop, this has caused perf issue for a disagg
+            # setup.
+            pinned = self.sampled_token_ids_pinned_cpu[:sampled_token_ids.
+                                                       shape[0]]
+            pinned.copy_(sampled_token_ids, non_blocking=True)
+            self.transfer_event.record()
+            self.transfer_event.synchronize()
```

Comment:
Would this work instead of using an event?
```suggestion
            pinned.copy_(sampled_token_ids, non_blocking=True)
            torch.cuda.current_stream().synchronize()
```

Reply:
From the observation, i think the CUDA stream synchronization is what has caused us the issue. I added a few more details in the issue ticket: https://github.com/vllm-project/vllm/issues/22754

Reply:
OK I was just curious whether the problem was that all streams were being synchronized.

If my suggestion doesn't work or is inferior in some way then the event synchronization looks good to me!

Reply:
> OK I was just curious whether the problem was that all streams were being synchronized.

Yes, I think this is what we have observed.